File: NetFramework/Charts/CS/CoordMapping3D/InputValueForm.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;

namespace CoordMapping
{
	/// <summary>
	/// Summary description for InputValueForm.
	/// </summary>
	public class InputValueForm : System.Windows.Forms.Form
	{
		private System.Windows.Forms.TextBox tbValue;
		private System.Windows.Forms.Button btnOk;
		private System.Windows.Forms.Button btnCancel;
		private System.Windows.Forms.Button btnHole;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public InputValueForm()
		{
			InitializeComponent();
		}

		public double ZValue
		{
			get
			{
				// return exactly the same hole value
				if( tbValue.Text.Equals("(Hole)"))
					return m_hole;
				try
				{
					return double.Parse( tbValue.Text);
				}
				catch
				{
					return m_hole;
				}
			}
			set
			{
				tbValue.Text = value.ToString();
			}
		}

		double m_hole = double.NaN;				
		public double Hole
		{
			get
			{
				return m_hole;
			}
			set
			{
				m_hole = value;
			}
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )//!!VBReplace-Overrides-Overloads Overrides-
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.tbValue = new System.Windows.Forms.TextBox();
			this.btnOk = new System.Windows.Forms.Button();
			this.btnCancel = new System.Windows.Forms.Button();
			this.btnHole = new System.Windows.Forms.Button();
			this.SuspendLayout();
			// 
			// tbValue
			// 
			this.tbValue.Location = new System.Drawing.Point(16, 16);
			this.tbValue.Name = "tbValue";
			this.tbValue.Size = new System.Drawing.Size(104, 20);
			this.tbValue.TabIndex = 0;
			this.tbValue.Text = "";
			// 
			// btnOk
			// 
			this.btnOk.DialogResult = System.Windows.Forms.DialogResult.OK;
			this.btnOk.Location = new System.Drawing.Point(16, 56);
			this.btnOk.Name = "btnOk";
			this.btnOk.Size = new System.Drawing.Size(75, 20);
			this.btnOk.TabIndex = 1;
			this.btnOk.Text = "OK";
			// 
			// btnCancel
			// 
			this.btnCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel;
			this.btnCancel.Location = new System.Drawing.Point(136, 56);
			this.btnCancel.Name = "btnCancel";
			this.btnCancel.Size = new System.Drawing.Size(75, 20);
			this.btnCancel.TabIndex = 2;
			this.btnCancel.Text = "Cancel";
			// 
			// btnHole
			// 
			this.btnHole.Location = new System.Drawing.Point(136, 16);
			this.btnHole.Name = "btnHole";
			this.btnHole.Size = new System.Drawing.Size(75, 20);
			this.btnHole.TabIndex = 3;
			this.btnHole.Text = "Set Hole";
			this.btnHole.Click += new System.EventHandler(this.btnHole_Click);
			// 
			// InputValueForm
			// 
			this.AcceptButton = this.btnOk;
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
			this.CancelButton = this.btnCancel;
			this.ClientSize = new System.Drawing.Size(226, 87);
			this.Controls.AddRange(new System.Windows.Forms.Control[] {
																																	this.btnHole,
																																	this.btnCancel,
																																	this.btnOk,
																																	this.tbValue});
			this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog;
			this.Name = "InputValueForm";
			this.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent;
			this.Text = "Enter Z-Value";
			this.ResumeLayout(false);

		}
		#endregion

		private void btnHole_Click(object sender, System.EventArgs e)
		{
			if( Hole!=double.NaN)
				tbValue.Text = "(Hole)";
		}
	}
}

File: NetFramework/Charts/VB/CoordMapping3D/InputValueForm.vb

Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms



'/ <summary>
'/ Summary description for InputValueForm.
'/ </summary>
Public Class InputValueForm
   Inherits System.Windows.Forms.Form
   Private tbValue As System.Windows.Forms.TextBox
   WithEvents btnOk As System.Windows.Forms.Button
   WithEvents btnCancel As System.Windows.Forms.Button
   WithEvents btnHole As System.Windows.Forms.Button
   '/ <summary>
   '/ Required designer variable.
   '/ </summary>
   Private components As System.ComponentModel.Container = Nothing
   
   
   Public Sub New()
      InitializeComponent()
   End Sub 'New
   
   
   Public Property ZValue() As Double
      Get
         ' return exactly the same hole value
         If tbValue.Text.Equals("(Hole)") Then
            Return m_hole
         End If
         Try
            Return Double.Parse(tbValue.Text)
         Catch
            Return m_hole
         End Try
      End Get
      Set
         tbValue.Text = value.ToString()
      End Set
   End Property
   
   Private m_hole As Double = Double.NaN
   
   Public Property Hole() As Double
      Get
         Return m_hole
      End Get
      Set
         m_hole = value
      End Set
   End Property
   
   
   '/ <summary>
   '/ Clean up any resources being used.
   '/ </summary>
   Protected Overloads Overrides Sub Dispose(disposing As Boolean) 
      If disposing Then
         If Not (components Is Nothing) Then
            components.Dispose()
         End If
      End If
      MyBase.Dispose(disposing)
   End Sub 'Dispose
   
   
#Region " Windows Form Designer generated code "
   '/ <summary>
   '/ Required method for Designer support - do not modify
   '/ the contents of this method with the code editor.
   '/ </summary>
   Private Sub InitializeComponent()
      Me.tbValue = New System.Windows.Forms.TextBox()
      Me.btnOk = New System.Windows.Forms.Button()
      Me.btnCancel = New System.Windows.Forms.Button()
      Me.btnHole = New System.Windows.Forms.Button()
      Me.SuspendLayout()
      ' 
      ' tbValue
      ' 
      Me.tbValue.Location = New System.Drawing.Point(16, 16)
      Me.tbValue.Name = "tbValue"
      Me.tbValue.Size = New System.Drawing.Size(104, 20)
      Me.tbValue.TabIndex = 0
      Me.tbValue.Text = ""
      ' 
      ' btnOk
      ' 
      Me.btnOk.DialogResult = System.Windows.Forms.DialogResult.OK
      Me.btnOk.Location = New System.Drawing.Point(16, 56)
      Me.btnOk.Name = "btnOk"
      Me.btnOk.Size = New System.Drawing.Size(75, 20)
      Me.btnOk.TabIndex = 1
      Me.btnOk.Text = "OK"
      ' 
      ' btnCancel
      ' 
      Me.btnCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel
      Me.btnCancel.Location = New System.Drawing.Point(136, 56)
      Me.btnCancel.Name = "btnCancel"
      Me.btnCancel.Size = New System.Drawing.Size(75, 20)
      Me.btnCancel.TabIndex = 2
      Me.btnCancel.Text = "Cancel"
      ' 
      ' btnHole
      ' 
      Me.btnHole.Location = New System.Drawing.Point(136, 16)
      Me.btnHole.Name = "btnHole"
      Me.btnHole.Size = New System.Drawing.Size(75, 20)
      Me.btnHole.TabIndex = 3
      Me.btnHole.Text = "Set Hole"
      ' 
      ' InputValueForm
      ' 
      Me.AcceptButton = Me.btnOk
      Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
      Me.CancelButton = Me.btnCancel
      Me.ClientSize = New System.Drawing.Size(226, 87)
      Me.Controls.AddRange(New System.Windows.Forms.Control() {Me.btnHole, Me.btnCancel, Me.btnOk, Me.tbValue})
      Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
      Me.Name = "InputValueForm"
      Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent
      Me.Text = "Enter Z-Value"
      Me.ResumeLayout(False)
   End Sub 'InitializeComponent
    
#End Region
   Private Sub btnHole_Click(sender As Object, e As System.EventArgs) Handles btnHole.Click
      If Hole <> Double.NaN Then
         tbValue.Text = "(Hole)"
      End If
   End Sub 'btnHole_Click
End Class 'InputValueForm
